src/realm/object-store/sync/sync_session.cpp

```diff
@@ -970,6 +1028,9 @@ void SyncSession::close(util::CheckedUniqueLock lock)
         case State::Dying:
             m_state_mutex.unlock(lock);
             break;
+        case State::Paused:
+            m_state_mutex.unlock(lock);
```

Comment:
Would you mind adding a comment here that mentions paused is supposed to be sticky and that state should not be changed to inactive.